### PR TITLE
Refactor the registry tests to operate on the plugin scope 

### PR DESCRIPTION
**Pull request by steveej**
- [x] Requires #313.

Split from #308.

### CODE REVIEW

**Comment by steveej** [comment]
/hold

- [x] #313 shall go in first.

File: e2e/tests/e2e.rs [hunk 1]
```diff
@@ -58,10 +58,14 @@ fn e2e_channel_success(channel: &'static str, arch: &'static str) {
...
-            "io.openshift.upgrades.graph.previous.remove",
-        ],
+        cincinnati::testing::CompareGraphsVerboseSettings {
+            unwanted_metadata_keys: &[
+                "io.openshift.upgrades.graph.previous.remove_regex",
+                "io.openshift.upgrades.graph.previous.remove",
+            ],
+
+            ..Default::default()
+        },
```
Comment (LalatenduMohanty) [inline e2e/tests/e2e.rs:68]: @steveeJ I am not sure what is this for? Why is this called `unwanted_metadata_keys`? Can you add more context to the commit?

Comment (LalatenduMohanty) [reply e2e/tests/e2e.rs:68]: This came from https://github.com/openshift/cincinnati/pull/312/commits/4cefba246adc9d37ef8965634c405e7a1b569a6d but still, a bit explanation/documentation/comment will help.

**Review by LalatenduMohanty — CHANGES_REQUESTED** [review]
We should squash https://github.com/openshift/cincinnati/pull/312/commits/ebceead1d7f448f4d0d07511202d8b2ed22657ad and  https://github.com/openshift/cincinnati/pull/312/commits/4cefba246adc9d37ef8965634c405e7a1b569a6d  as there are essentially related code changes.

File: e2e/tests/e2e.rs (hunk 1 above)
Comment (steveej) [reply e2e/tests/e2e.rs:68]: It effectively allows us to omit arguments to the `graph_compare_graph` and leaving them at their default.

Comment (steveej) [reply e2e/tests/e2e.rs:68]: We don't want to include metadata with the given keys in the graph comparison, so we remove these before comparing the graphs.

**Comment by steveej** [comment]
> We should squash [ebceead](https://github.com/openshift/cincinnati/commit/ebceead1d7f448f4d0d07511202d8b2ed22657ad) and [4cefba2](https://github.com/openshift/cincinnati/commit/4cefba246adc9d37ef8965634c405e7a1b569a6d) as there are essentially related code changes.

All comments addressed, thanks for the review!

File: cincinnati/src/lib.rs [hunk 2]
```diff
@@ -1017,6 +1040,45 @@ pub mod testing {
...
+                            Error::msg(format!("invalid version string {:?}", version))
+                        })?;
+                        release.payload = format!("{}:{}", source_front, version);
+
+                        Ok(())
+                    }
+                    _ => panic!("should not get here"),
+                }
+            })
+            .context("Post-processing the received graph to match the test expectations")
```
Comment (LalatenduMohanty) [inline cincinnati/src/lib.rs]: This message `Post-processing the received graph to match the test expectations` is only specific to tests and does not sound right.

File: cincinnati/src/lib.rs [hunk 3]
```diff
@@ -1017,6 +1040,45 @@ pub mod testing {
...
+                        // replace digest by tag to match expectency
+                        let version = release.version.to_string();
+                        release.payload = version;
+
+                        Ok(())
+                    }
+                    _ => panic!("should not get here"),
+                }
+            })
+            .context("Post-processing the received graph to match the test expectations")
```
Comment (LalatenduMohanty) [inline cincinnati/src/lib.rs]: same here. https://github.com/openshift/cincinnati/pull/312/commits/02f94bc9d668970731d224b9db755f3cbf878e03#r515120447

File: cincinnati/src/lib.rs [hunk 4]
```diff
@@ -1017,6 +1040,45 @@ pub mod testing {
...
+            .iter_releases_mut(|ref mut release| {
+                match release {
+                    Release::Concrete(ref mut release) => {
+                        // replace digest by tag to match expectency
+                        let version = release.version.to_string();
+                        release.payload = version;
+
+                        Ok(())
+                    }
+                    _ => panic!("should not get here"),
```
Comment (LalatenduMohanty) [inline cincinnati/src/lib.rs]: Another console message which I think should be modified to be more meaningful. Forgot to point this out in last review.

**Review by LalatenduMohanty — CHANGES_REQUESTED** [review]

File: cincinnati/src/lib.rs [hunk 5]
```diff
@@ -1017,6 +1040,45 @@ pub mod testing {
...
+                        // replace digest by tag to match expectency
+                        let version = release.version.to_string();
+                        let source_front = release.payload.split('@').next().ok_or_else(|| {
+                            Error::msg(format!("invalid version string {:?}", version))
+                        })?;
+                        release.payload = format!("{}:{}", source_front, version);
+
+                        Ok(())
+                    }
+                    _ => panic!("should not get here"),
```
Comment (LalatenduMohanty) [inline cincinnati/src/lib.rs]: https://github.com/openshift/cincinnati/pull/312/files#r515210503

**Review by LalatenduMohanty — APPROVED** [review]
/lgtm

**Pull request merged**